src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -209,18 +210,27 @@ internal JsonPropertyInfo CreateRootObject(JsonSerializerOptions options)
                 options: options);
         }
 
-        internal JsonPropertyInfo CreatePolymorphicProperty(JsonPropertyInfo property, Type runtimePropertyType, JsonSerializerOptions options)
+        internal JsonPropertyInfo GetOrAddPolymorphicProperty(JsonPropertyInfo property, Type runtimePropertyType, JsonSerializerOptions options)
         {
-            JsonPropertyInfo runtimeProperty = CreateProperty(
-                property.DeclaredPropertyType, runtimePropertyType,
-                property.ImplementedPropertyType,
-                property.PropertyInfo,
-                parentClassType: Type,
-                converter: null,
-                options: options);
-            property.CopyRuntimeSettingsTo(runtimeProperty);
+            static JsonPropertyInfo CreateRuntimeProperty((JsonPropertyInfo property, Type runtimePropertyType) key, (JsonSerializerOptions options, Type classType) arg)
+            {
+                JsonPropertyInfo runtimeProperty = CreateProperty(
+                    key.property.DeclaredPropertyType, key.runtimePropertyType,
+                    key.property.ImplementedPropertyType,
+                    key.property.PropertyInfo,
+                    parentClassType: arg.classType,
+                    converter: null,
+                    options: arg.options);
+
+                key.property.CopyRuntimeSettingsTo(runtimeProperty);
+
+                return runtimeProperty;
+            }
+
+            ConcurrentDictionary<(JsonPropertyInfo, Type), JsonPropertyInfo> cache =
+                LazyInitializer.EnsureInitialized(ref RuntimePropertyCache, () => new ConcurrentDictionary<(JsonPropertyInfo, Type), JsonPropertyInfo>());
 
-            return runtimeProperty;
+            return cache.GetOrAdd((property, runtimePropertyType), (key, arg) => CreateRuntimeProperty(key, arg), (options, Type));
```

Comment:
Specify the generic explicitly?
```suggestion
            return cache.GetOrAdd<(JsonSerializerOptions, Type)>((property, runtimePropertyType), (key, arg) => CreateRuntimeProperty(key, arg), (options, Type));
```

Reply:
This API isn't available on netstandard 2.0, which is why the build is failing.
Can you #if/def the code using BUILDING_INBOX_LIBRARY and in the else block, only use netstandard2.0 compatible APIs?